app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,78 @@
+class CategoriesController < ApplicationController
+
+  before_action :require_login
+
+  def index
+    @categories = Category.all
+    @user = User.find(session[:user_id])
+  end
+
+  def new
+    user = User.find(session[:user_id])
+    if user.admin.eql?(1)
```

Comment:
if we're using enum, then we have access to a method called `admin?` so we don't have to manually check the value of this attribute. if we aren't using enum, i suggest writing a method in your `User` method that's called `admin?`:

```ruby
def admin?
  self.admin == 1
end
```